File: packages/shared-types/src/resources/digest.ts
export interface DigestEntryResource {
  cluster_id: string;
  rank: number;
  category: string;
  topic?: string;
  headline: string;
  summary: string;
  source_count: number;
  sources?: string[];
  image_url?: string;
  video_url?: string;
  headline_translated?: string;
  summary_translated?: string;
}

export interface DigestResource {
  date: string;
  published_at: string;
  entries: DigestEntryResource[];
}

File: services/api/app/schemas/digest.py
from pydantic import BaseModel


class DigestEntryResource(BaseModel):
    cluster_id: str
    rank: int
    category: str
    headline: str
    summary: str
    source_count: int
    sources: list[str] = []
    image_url: str = ""
    video_url: str = ""
    headline_translated: str = ""
    summary_translated: str = ""


class DigestResource(BaseModel):
    date: str
    published_at: str
    entries: list[DigestEntryResource]
